crates/ty_server/src/server/api/requests/completion.rs

```diff
@@ -69,3 +71,42 @@ impl BackgroundDocumentRequestHandler for CompletionRequestHandler {
 impl RetriableRequestHandler for CompletionRequestHandler {
     const RETRY_ON_CANCELLATION: bool = true;
 }
+
+fn kind(ty: Type<'_>) -> Option<CompletionItemKind> {
+    // ref https://microsoft.github.io/language-server-protocol/specifications/lsp/3.17/specification/#completionItemKind
+    Some(match ty {
+        Type::FunctionLiteral(_)
+        | Type::DataclassDecorator(_)
+        | Type::DataclassTransformer(_)
+        | Type::Callable(_) => CompletionItemKind::FUNCTION,
+        Type::BoundMethod(_) => CompletionItemKind::METHOD,
+        Type::ModuleLiteral(_) => CompletionItemKind::MODULE,
+        Type::ClassLiteral(_) | Type::GenericAlias(_) | Type::SubclassOf(_) => {
+            CompletionItemKind::CLASS
+        }
+        // This is a little weird for "struct." I'm mostly interpreting
+        // "struct" here as a more general "object." ---AG
+        Type::NominalInstance(_)
+        | Type::ProtocolInstance(_)
+        | Type::PropertyInstance(_)
```

Comment:
There seems to be a `CompletionItemKind::PROPERTY`, should we use that instead here?

Reply:
I guess `PROPERTY` was probably meant for something else. It's also extremely unlikely that someone will see something of type `PropertyInstance`, as you would normally see it's resolved `__get__` return type.